pandas/io/formats/style_render.py

```diff
@@ -854,13 +859,11 @@ def concatenated_visible_rows(obj, n, row_indices):
             row_indices.extend(
                 [r + n for r in range(len(obj.index)) if r not in obj.hidden_rows]
             )
-            return (
-                row_indices
-                if obj.concatenated is None
-                else concatenated_visible_rows(
-                    obj.concatenated, n + len(obj.index), row_indices
+            for concatenated in obj.concatenated:
+                row_indices = concatenated_visible_rows(
```

Comment:
are you not going to face the same issue here. You are overwriting `row_indices` from those of the last concatenated styler in the list, and ignoring all prior ones in the list?